geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientProxyMembershipID.java

```diff
@@ -176,149 +177,114 @@ public static ClientProxyMembershipID getClientId(DistributedMember member) {
+          ((InternalDistributedMember) systemMemberId).setDurableId(attributes.getId());
         }
-        DataSerializer.writeObject(systemMemberId, hdos);
-        client_side_identity = hdos.toByteArray();
-      } catch (IOException ioe) {
-        throw new InternalGemFireException(
-            "Unable to serialize identity",
-            ioe);
       }
-
-      system = sys;
+      DataSerializer.writeObject(systemMemberId, hdos);
+      client_side_identity = hdos.toByteArray();
+    } catch (IOException ioe) {
+      throw new InternalGemFireException("Unable to serialize identity", ioe);
     }
+    system = sys;
     return client_side_identity;
-
   }
 
   private ClientProxyMembershipID(int id, byte[] clientSideIdentity) {
-    boolean specialCase = Boolean.getBoolean(GeodeGlossary.GEMFIRE_PREFIX + "SPECIAL_DURABLE");
-    String durableID = this.system.getProperties().getProperty(DURABLE_CLIENT_ID);
+    this(clientSideIdentity, getUniqueId(id), systemMemberId);
+  }
+
+  private static int getUniqueId(final int id) {
+    final boolean specialCase =
+        Boolean.getBoolean(GeodeGlossary.GEMFIRE_PREFIX + "SPECIAL_DURABLE");
+    final String durableID = system.getProperties().getProperty(DURABLE_CLIENT_ID);
     if (specialCase && durableID != null && (!durableID.equals(""))) {
-      this.uniqueId = durable_synch_counter;
+      return durable_synch_counter;
     } else {
-      this.uniqueId = id;
+      return id;
     }
-    this.identity = clientSideIdentity;
-    this.memberId = systemMemberId;
   }
 
   public ClientProxyMembershipID() {}
 
-  public ClientProxyMembershipID(DistributedMember member) {
-    this.uniqueId = 1;
-    this.memberId = member;
+  public ClientProxyMembershipID(final DistributedMember member) {
+    this(null, 1, member);
     updateID(member);
   }
 
+  @VisibleForTesting
+  ClientProxyMembershipID(final byte[] identity, final int uniqueId,
+      final DistributedMember memberId) {
+    this.identity = identity;
+    this.uniqueId = uniqueId;
+    this.memberId = memberId;
+  }
 
   private transient String _toString;
 
-  // private transient int transientPort; // variable for debugging member ID issues
-
   @Override
   public String toString() {
-    if (this.identity != null
+    if (identity != null
         && ((InternalDistributedMember) getDistributedMember()).getMembershipPort() == 0) {
-      return this.toStringNoCache();
+      return toStringNoCache();
     }
-    if (this._toString == null) {
-      this._toString = this.toStringNoCache();
+    if (_toString == null) {
+      _toString = toStringNoCache();
     }
-    return this._toString;
+    return _toString;
   }
 
   /**
    * returns a string representation of this identifier, ignoring the toString cache
    */
   public String toStringNoCache() {
-    StringBuffer sb = new StringBuffer("identity(").append(getDSMembership()).append(",connection=")
-        .append(uniqueId);
+    StringBuilder sb =
```

Comment:
request: No need for a `StringBuilder` here. If you're bothering to change this code I recommend changing it to use the `+` operator.

Reply:
(see conversation on other `StringBuilder`) I'm resolving this comment because this is called in a `toString()` and who knows, maybe it can affect performance.